### PR TITLE
Cpp creation

### PR DESCRIPTION
**Pull request by LiorB-D**
Initial commit

### CODE REVIEW

**Comment by enki-hq** [comment]
### Workout

| Workout Name | Error Message |
|:-:|:-:|
| intro-to-cpp | Workout description must be a non-empty string. |
| logical-operators-cpp | Workout description must be a non-empty string. |
| more-data-types-cpp | Workout description must be a non-empty string. |
| looping-cpp | One of the mentioned insights couldn't be found in this pull request or the database. |
| control-flow-i-cpp | One of the mentioned insights couldn't be found in this pull request or the database. |
### Course

| Course Name | Error Message |
|:-:|:-:|
| cpp-core | Course is missing its description. |
### Topic

| Topic Name | Error Message |
|:-:|:-:|
| cpp | Topic color `null` is not valid hex |
| cpp | Topic description must be a non-empty string. |

> If you notice anything wrong with these rules contact content-team@enki.com.

**Comment by enki-hq** [comment]
### Workout

| Workout Name | Error Message |
|:-:|:-:|
| control-flow-i-cpp | One of the mentioned insights couldn't be found in this pull request or the database. |
| looping-cpp | One of the mentioned insights couldn't be found in this pull request or the database. |
### Course

| Course Name | Error Message |
|:-:|:-:|
| cpp-core | Course is missing its description. |
### Topic

| Topic Name | Error Message |
|:-:|:-:|
| cpp | Topic color `null` is not valid hex |
| cpp | Topic description must be a non-empty string. |

> If you notice anything wrong with these rules contact content-team@enki.com.

**Comment by enki-hq** [comment]
### Course

| Course Name | Error Message |
|:-:|:-:|
| cpp-core | Course is missing its description. |
### Topic

| Topic Name | Error Message |
|:-:|:-:|
| cpp | Topic color `#044F88` does not conform to our HSV rules. Replace it with `#a6d8ff`. |

> If you notice anything wrong with these rules contact content-team@enki.com.

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/cpp/cpp-core/control-flow-i-cpp/comparison-operators-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/control-flow-i-cpp/comparison-operators-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/control-flow-i-cpp/if-else-statements-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/control-flow-i-cpp/if-else-statements-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/control-flow-i-cpp/if-statements-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/control-flow-i-cpp/if-statements-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/control-flow-i-cpp/nested-if-else-statements-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/control-flow-i-cpp/nested-if-else-statements-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/intro-to-cpp/first-program-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/intro-to-cpp/first-program-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/intro-to-cpp/installing-cpp-locally.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/intro-to-cpp/installing-cpp-locally.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/intro-to-cpp/what-is-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/intro-to-cpp/what-is-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/intro-to-cpp/why-learn-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/intro-to-cpp/why-learn-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/logical-operators-cpp/and-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/logical-operators-cpp/and-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/logical-operators-cpp/not-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/logical-operators-cpp/not-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/logical-operators-cpp/or-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/logical-operators-cpp/or-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/logical-operators-cpp/what-are-logical-operators-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/logical-operators-cpp/what-are-logical-operators-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/looping-cpp/do-while-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/looping-cpp/do-while-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/looping-cpp/for-loops-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/looping-cpp/for-loops-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/looping-cpp/what-are-loops-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/looping-cpp/what-are-loops-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/looping-cpp/while-loops-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/looping-cpp/while-loops-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/more-data-types-cpp/arrays-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/more-data-types-cpp/arrays-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/more-data-types-cpp/vector-methods-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/more-data-types-cpp/vector-methods-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/more-data-types-cpp/vectors-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/more-data-types-cpp/vectors-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/assignment-operators-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/assignment-operators-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/basic-data-types-cpp-ii.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/basic-data-types-cpp-ii.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/basic-data-types-cpp-iii.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/basic-data-types-cpp-iii.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/basic-data-types-cpp-iv.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/basic-data-types-cpp-iv.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/basic-data-types-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/basic-data-types-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/creating-and-storing-variables-in-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/creating-and-storing-variables-in-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/printing-variables-in-cpp-ii.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/printing-variables-in-cpp-ii.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)
- [ ] [enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/printing-variables-in-cpp.md](https://app.enki.com/public/preview-insight/enkidevs/curriculum/cpp/cpp-core/variables-and-data-types-cpp/printing-variables-in-cpp.md?ref=603721f04d3dd90271fda4ebac233df2f90bd62e)

**Comment by enki-hq** [comment]
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| installing-cpp-locally | Link checking took too long. Verify it manually. |

> If you notice anything wrong with these rules contact content-team@enki.com.

**Comment by enki-hq** [comment]
Failed parsing files: 
  - cpp/cpp-core/control-flow-i-cpp/nested-if-else-statements-cpp.md: 
 > Missing compiler for node of type `questionGap`: `[object Object]`
  - cpp/cpp-core/looping-cpp/while-loops-cpp.md: 
 > Missing compiler for node of type `questionCode`: `[object Object]`

**Comment by enki-hq** [comment]
Failed parsing files: 
  - cpp/cpp-core/control-flow-i-cpp/nested-if-else-statements-cpp.md: 
 > Missing compiler for node of type `questionCode`: `[object Object]`

File: cpp/cpp-core/intro-to-cpp/assumed-knowledge-preface.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Content
+
+Ready for a thrilling dive into C++? 
+
+It's like the action hero of programming languages - complex but oh-so-powerful.  💪
+
+If 'for-loop', 'variable', 'arrays', and 'functions' are part of your vocabulary, you're all set to jump in!
+
+But if those terms made you go, "Wait, what? 🤔", you should take a detour to our [Intro to Coding](https://app.enki.com/skill/coding-intro).
```
Comment (nem035) [inline cpp/cpp-core/intro-to-cpp/assumed-knowledge-preface.md]: 👍

**Comment by enki-hq** [comment]
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| printing-variables-in-cpp | No footnote matches found in content. |

> If you notice anything wrong with these rules contact content-team@enki.com.

**Comment by enki-hq** [comment]
### Workout

| Workout Name | Error Message |
|:-:|:-:|
| data-types | Workout slug `data-types` isn't unique in the database. |

> If you notice anything wrong with these rules contact content-team@enki.com.

**Comment by enki-hq** [comment]
### Insight

| Insight Name | Error Message |
|:-:|:-:|
| variables-case-sensitive | Insight doesn't have a practice question but it has formats and context for it. |

> If you notice anything wrong with these rules contact content-team@enki.com.

**Comment by enki-hq** [comment]
### Course

| Course Name | Error Message |
|:-:|:-:|
| cpp-core | We couldn't find these workouts: `data-types-cpp`. |

> If you notice anything wrong with these rules contact content-team@enki.com.

**Review by nem035 — APPROVED** [review]

**Pull request merged**